include/networkit/edgescores/EdgeScoreAsWeight.hpp

```diff
@@ -12,7 +12,9 @@
 
 namespace NetworKit {
 
-class EdgeScoreAsWeight final {
+class [[deprecated(
```

Comment:
Better use `TLX_DEPRECATED` for deprecating the class (see [here](https://github.com/networkit/networkit/blob/e326d93121f699902266c64ff2bbd1b16c6e61e6/include/networkit/graph/UnionMaximumSpanningForest.hpp#L62) for an example used for a method.)

In addition, add a DEPRECATION hint in the documentation of the constructor.